File: office-space-backend/src/main/java/com/ourfancyteamname/officespace/services/PackageService.java
package com.ourfancyteamname.officespace.services;

import org.springframework.data.domain.Page;

import com.ourfancyteamname.officespace.db.entities.Package;
import com.ourfancyteamname.officespace.db.entities.view.PackageListView;
import com.ourfancyteamname.officespace.dtos.PackageDto;
import com.ourfancyteamname.officespace.dtos.TableSearchRequest;

public interface PackageService {

  Package create(PackageDto productDto);

  Package update(PackageDto packageDto);

  void delete(Integer packageId);

  Page<PackageListView> getListView(TableSearchRequest tableSearchRequest);

}

File: office-space-frontend/src/main/web/office-space-web/src/app/services/package.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { TableSearchRequest } from '../dtos/tableSearch';
import { StorageService } from './auth/storage.service';

@Injectable({
  providedIn: 'root'
})
export class PackageService {

  readonly url = `${this.storage.get(StorageService.API)}package`;

  constructor(private http: HttpClient, private storage: StorageService) { }

  create(data): Observable<any> {
    return this.http.post(`${this.url}/create`, data);
  }

  getListView(data: TableSearchRequest): Observable<any> {
    return this.http.post(`${this.url}/list-view`, data);
  }

  update(data): Observable<any> {
    return this.http.patch(`${this.url}/update`, data);
  }

  delete(id: number): Observable<any> {
    return this.http.delete(`${this.url}/${id}`)
  }
}
